docs/source/typed_dict.rst

```diff
@@ -236,6 +235,68 @@ another ``TypedDict`` if all required keys in the other ``TypedDict`` are requir
 first ``TypedDict``, and all non-required keys of the other ``TypedDict`` are also non-required keys
 in the first ``TypedDict``.
 
+Depending on the totality of the ``TypedDict`` either ``Required`` or
+``NotRequired`` can be explicitly used on some of the keys. For instance, the
+``Movie`` type from the example above could be also defined as:
+
+.. code-block:: python
+
+   from typing_extensions import Required
+
+   class Movie(TypedDict, total=False):
+       name: Required[str]
+       year: Required[int]
+       based_on: str
+
+and the two definitions would be equivalent.
+
+As a rule of thumb, if more keys of a particular ``TypedDict`` are required
+than not, construct the new type with the ``total`` parameter set to ``True``
+and qualify the non required keys using ``NotRequired``. Otherwise, construct
+the new type with the ``total`` parameter set to ``False`` and qualify the
+required keys with ``Required``.
+
+Using ``Required`` with a total ``TypedDict`` and using ``NotRequired`` with a
+partial ``TypedDict`` is redundant. However, it is allowed and can be used to
+qualify the required and not required keys explicitly.
+
+If a particular key can accept ``None``, it is recommended to avoid mixing
+``Optional`` with either ``Required`` or ``NotRequired`` and use the
+``TYPE|None`` notation instead:
+
+.. code-block:: python
+
+   # Preferred approach.
+   class Car(TypedDict):
+       model: str
+       owner: NotRequired[str | None]
+
+   # Not recommended.
+   class Car(TypedDict):
+       model: str
+       owner: NotRequired[Optional[str]]
```

Comment:
Let me know if this is too long. Maybe we could just link to the "[How to teach this" section of the PEP 655](https://peps.python.org/pep-0655/)?